rocksdb_admin/admin_handler.cpp

```diff
@@ -1101,29 +1126,21 @@ void AdminHandler::async_tm_stopMessageIngestion(
     return;
   }
 
-  std::shared_ptr<KafkaWatcher> kafka_watcher;
-  {
-    std::lock_guard<std::mutex> lock(kafka_watcher_lock_);
-    auto iter = kafka_watcher_map_.find(db_name);
-    // Verify that there is thread consuming messages for this db.
-    if (iter == kafka_watcher_map_.end()) {
+  if (!kafka_spec_manager_.isSpecPresent(db_name)) {
       e.message = db_name + " is not being consumed";
       callback.release()->exceptionInThread(std::move(e));
       LOG(ERROR) << db_name << " is not being currently consumed";
       return;
-    }
-    kafka_watcher = iter->second;
   }
 
+  auto kafka_watcher = kafka_spec_manager_.getKafkaWatcher(db_name);
```

Comment:
is it possible here to write the message that we last consumed? It would eliminate the gap between when the function was last run and the last message we've actually consumed

Reply:
Yes, I've added this. Thanks.